### PR TITLE
INT B-18824 requested office users get single user endpoint

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

***please ignore my brainfart*** the branch has the incorrect BL item, it should be `B-18824` and not `B-18885`

When it comes to having office users requesting their account, the admins will eventually need to review that user in the Admin console within MilMove. This will require an endpoint that can be called to retrieve that information so the UI can display it. I took notes off of what was being done for viewing admin users and pretty much copied that functionality over to `requested_office_users`.

I have merged the branch of [B-18822](https://github.com/transcom/mymove/pull/12113) into this in order to copy over the creation of the endpoint done in that branch.

The return value of querying the db should look like this:
```
{
  "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
  "userId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "firstName": "string",
  "middleInitials": "string",
  "lastName": "string",
  "email": "dkj-9pUmr@uLdkqVEkf9MsW9h8yoezuK-Eg5HEirPkavMgiFw8tvPyuVMaKQy3uys3ig3IrQM5WOyg9x2yvOsc3K1M2kv.rBrA5hihPdWu3.kYiMtccDklvvtbpSBUutDabBbbDQwVkCdVbTDmiKssUrhyvBpMGdnTAgvrsTONFhcPznXnAPeydXIKvZmBFqBYobvctock",
  "telephone": "468-681-8762",
  "transportationOfficeId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "active": true,
  "roles": [
    {
      "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
      "roleType": "customer",
      "roleName": "Transportation Ordering Officer",
      "createdAt": "2024-03-01T14:36:30.952Z",
      "updatedAt": "2024-03-01T14:36:30.952Z"
    }
  ],
  "edipi": "string",
  "otherUniqueId": "string",
  "rejectionReason": "string",
  "status": "APPROVED",
  "createdAt": "2024-03-01T14:36:30.952Z",
  "updatedAt": "2024-03-01T14:36:30.952Z"
}
```

This PR does the following:
- adds `GET requested-office-users/{officeUserId}` endpoint that returns a single row in the `office_users` table
- updates `swagger-def` to include the endpoint and handle the return values
- updated and added relevant tests

### How to test

1. Fire up `client` & `server`
2. Access MM as an admin
3. It isn't necessary, but make sure you have some users in the `office_users` table, they don't NEED to have a `REQUESTED` status, this query should still return a user
4. Head on over to `http://adminlocal:3000/swagger-ui/admin.html#/`
5. Copy an office user's `id` from the `office_users` table
6. Find the `requested-office-users/{officeUserId}` endpoint and plug in that `id`
7. Execute that sucker and you should get a return value back

## Screenshots
Screenshot 2024-03-01 at 8 36 41 AM

### CODE REVIEW

**Review by traskowskycaci — APPROVED** [review]
Got a 200 response with this, which matches what's in the database. I did have to manually add an edipi to see that in the response:
```
{
  "active": true,
  "createdAt": "2024-03-01T21:19:32.632Z",
  "edipi": "1234567",
  "email": "too_role@office.mil",
  "firstName": "Leo",
  "id": "144503a6-485c-463e-b943-d3c3bad11b09",
  "lastName": "Spaceman",
  "middleInitials": null,
  "otherUniqueId": null,
  "rejectionReason": null,
  "roles": null,
  "status": null,
  "telephone": "415-555-1212",
  "transportationOfficeId": "7730c086-18ed-403f-800f-929e1839cad3",
  "updatedAt": "2024-03-01T21:19:32.632Z",
  "userId": "dcf86235-53d3-43dd-8ee8-54212ae3078f"
}
```

**Comment by danieljordan-caci** [comment]
> Got a 200 response with this, which matches what's in the database. I did have to manually add an edipi to see that in the response:
> 
> ```
> {
>   "active": true,
>   "createdAt": "2024-03-01T21:19:32.632Z",
>   "edipi": "1234567",
>   "email": "too_role@office.mil",
>   "firstName": "Leo",
>   "id": "144503a6-485c-463e-b943-d3c3bad11b09",
>   "lastName": "Spaceman",
>   "middleInitials": null,
>   "otherUniqueId": null,
>   "rejectionReason": null,
>   "roles": null,
>   "status": null,
>   "telephone": "415-555-1212",
>   "transportationOfficeId": "7730c086-18ed-403f-800f-929e1839cad3",
>   "updatedAt": "2024-03-01T21:19:32.632Z",
>   "userId": "dcf86235-53d3-43dd-8ee8-54212ae3078f"
> }
> ```

Looks good. Yeah, until we have the `PATCH` endpoint, most data will need to be manually changed.

File: swagger-def/admin.yaml [hunk 1]
```diff
@@ -1142,6 +1142,34 @@ paths:
...
           description: server error
+  /requested-office-users/{officeUserId}:
+    get:
+      produces:
+        - application/json
+      summary: Get a Requested Office User
+      description: Retrieving a single office user that is in a REQUESTED status
+      operationId: getRequestedOfficeUser
+      tags:
+        - Requested office users
```
Comment (cameroncaci) [inline swagger-def/admin.yaml:1153]: This API description make it out to be that this endpoint is *specifically* for "REQUESTED" status office users, which is not the case. Any status works

Comment (cameroncaci) [reply swagger-def/admin.yaml:1153]: Should be a small change while Daniel is out @deandreJones

**Review by cameroncaci — CHANGES_REQUESTED** [review]

File: swagger-def/admin.yaml (hunk 1 above)
Comment (deandreJones) [reply swagger-def/admin.yaml:1153]: should it only return users in requested status- or just return the user no matter the status - i would think that any is ok

Comment (deandreJones) [reply swagger-def/admin.yaml:1153]: looking at the AC- i think i have my answer- just need to adjust the description of this

Comment (danieljordan-caci) [reply swagger-def/admin.yaml]: @cameroncaci this has been updated

**Review by cameroncaci — APPROVED** [review]

**Pull request merged**